tests/python_tests/helpers/device.py

```diff
@@ -73,30 +81,54 @@ def write_stimuli_to_l1(buffer_A, buffer_B, stimuli_format, core_loc="0,0", tile
 
 
 def get_result_from_device(
-    format: str, read_data_bytes: bytes, core_loc: str = "0,0", sfpu: bool = False
+    formats: FormatConfig,
+    read_data_bytes: bytes,
+    core_loc: str = "0,0",
+    sfpu: bool = False,
 ):
     # Dictionary of format to unpacking function mappings
     unpackers = {
-        "Float16": unpack_fp16,
-        "Float16_b": unpack_bfp16,
-        "Float32": unpack_float32,
-        "Int32": unpack_int32,
+        DataFormat.Float16: unpack_fp16,
+        DataFormat.Float16_b: unpack_bfp16,
+        DataFormat.Float32: unpack_float32,
+        DataFormat.Int32: unpack_int32,
     }
 
     # Handling "Bfp8_b" format separately with sfpu condition
-    if format == "Bfp8_b":
+    if formats.pack_dst == DataFormat.Bfp8_b:
         unpack_func = unpack_bfp16 if sfpu else unpack_bfp8_b
     else:
-        unpack_func = unpackers.get(format)
+        unpack_func = unpackers.get(formats.pack_dst)
 
     if unpack_func:
-        return unpack_func(read_data_bytes)
+        num_args = len(inspect.signature(unpack_func).parameters)
+        if num_args > 1:
+            return unpack_func(
+                read_data_bytes, formats.unpack_src, formats.pack_dst
+            )  # Bug patchup in (unpack.py): in case unpack_src is Bfp8_b != pack_dst, L1 must be read in different order to extract correct results
+        else:
+            return unpack_func(read_data_bytes)
     else:
-        raise ValueError(f"Unsupported format: {format}")
+        raise ValueError(f"Unsupported format: {formats.pack_dst}")
+
+
+def check_value_with_timeout(core_loc, mailbox_addr, timeout=0, poll_interval=0.1):
+    start_time = time.time()
+
+    while time.time() - start_time < timeout:
```

Comment:
```suggestion
    end_time = time.time() + timeout

    while time.time() < end_time:
```
Seems easier to read.